src/event/event_kevent.c

```diff
@@ -1271,6 +1294,8 @@ _dispatch_event_loop_timer_program(uint32_t tidx,
 #endif
 	};
 
+	(void) leeway; // if DISPATCH_HAVE_TIMER_COALESCING == 0
```

Comment:
This change seems odd ...

Reply:
This is for `-Werror` compliance. Maybe it should be:

```
DISPATCH_NOINLINE
static void
_dispatch_event_loop_timer_program(uint32_t tidx,
		uint64_t target, uint64_t leeway, uint16_t action)
{
	dispatch_kevent_s ke = {
		.ident = DISPATCH_KEVENT_TIMEOUT_IDENT_MASK | tidx,
		.filter = EVFILT_TIMER,
		.flags = action | EV_ONESHOT,
		.fflags = _dispatch_timer_index_to_fflags[tidx],
		.data = (int64_t)target,
		.udata = (dispatch_kqueue_udata)&_dispatch_timers_heap[tidx],
#if DISPATCH_HAVE_TIMER_COALESCING
		.ext[1] = leeway,
#endif
#if DISPATCH_USE_KEVENT_QOS
		.qos = _PTHREAD_PRIORITY_EVENT_MANAGER_FLAG,
#endif
	};

#if !DISPATCH_HAVE_TIMER_COALESCING
	(void) leeway;
#endif

	_dispatch_kq_deferred_update(DISPATCH_WLH_ANON, &ke);
}
```

(added the whole function for more context)

Reply:
no the change isn't odd, this is because leeway is unused, and it's fine to do it in an unguarded part. `(void)foo;` to avoid an unused warning is idiomatic in libdispatch.